builder/haddock-builder.nix

```diff
@@ -0,0 +1,132 @@
+{ stdenv, buildPackages, lib, haskellLib, ghc, ghcForComponent, nonReinstallablePkgs, runCommand, writeText, writeScript, makeConfigFiles }:
+
+{ componentId
+, component
+, package
+, flags
+, revision
+, commonAttrs
+, configureFlags
+
+, doHoogle
+, hyperlinkSource
+, setupHaddockFlags
+, setupGhcOptions
+
+, needsProfiling
+, componentDrv
+, configFiles # component config files
+}:
+
+let
+  # the target dir for haddock documentation
+  docdir = docoutput: docoutput + "/share/doc/" + componentId.cname;
+
+  packageCfgDir = configFiles.packageCfgDir;
+
+  fullName = "${componentDrv.name}-haddock";
+
+  docsConfigFiles = makeConfigFiles {
+    inherit (package) identifier;
+    inherit component fullName flags needsProfiling;
+    chooseDrv = p: p.haddock;
+  };
+
+  finalConfigureFlags = lib.concatStringsSep " " (
```

Comment:
Does anything go wrong if we share the configure flags for the main component and just add the haddock bits?

Reply:
We need the modified `haddock-interfaces:` for symlinks to be constructed correctly and we can't know at build time the path to use in `haddock-interfaces:`.